plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/converter/GlueStatConverter.java

```diff
@@ -0,0 +1,315 @@
+import com.amazonaws.services.glue.model.LongColumnStatisticsData;
+import com.amazonaws.services.glue.model.StringColumnStatisticsData;
+import io.trino.plugin.hive.HiveType;
+import io.trino.plugin.hive.metastore.Column;
+import io.trino.plugin.hive.metastore.HiveColumnStatistics;
+import io.trino.plugin.hive.metastore.Partition;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.spi.TrinoException;
+import org.apache.hadoop.hive.metastore.api.Decimal;
+import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalLong;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.plugin.hive.HiveErrorCode.HIVE_INVALID_METADATA;
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createBinaryColumnStatistics;
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createBooleanColumnStatistics;
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createDateColumnStatistics;
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createDecimalColumnStatistics;
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createDoubleColumnStatistics;
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createIntegerColumnStatistics;
+import static io.trino.plugin.hive.metastore.HiveColumnStatistics.createStringColumnStatistics;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreUtil.fromMetastoreDistinctValuesCount;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreUtil.fromMetastoreNullsCount;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreUtil.getAverageColumnLength;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreUtil.getTotalSizeInBytes;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreUtil.toMetastoreDistinctValuesCount;
+import static org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector.Category.PRIMITIVE;
+
+public class GlueStatConverter
+{
+    private GlueStatConverter() {}
+
+    private static final long MILLIS_PER_DAY = TimeUnit.DAYS.toMillis(1);
+
+    public static List<ColumnStatistics> toGlueColumnStatistics(
+            Partition partition, Map<String, HiveColumnStatistics> trinoColumnStats, OptionalLong rowCount)
+    {
+        List<ColumnStatistics> catalogColumnStatisticsList = new ArrayList<>(trinoColumnStats.size());
+
+        trinoColumnStats.forEach((columnName, statistics) -> {
+            final Optional<Column> column = columnByName(partition, columnName);
```

Comment:
this is O(N^2) which with 1000 columns will start to be substantial cost. Pre-build String->Column map

Reply:
Naa - it is only partition columns. So I guess we will not have 1000 of those :)

Reply:
Still should be easy to remedy and there's no reason to avoid this unless it's more difficult than expected - https://github.com/trinodb/trino/pull/6178#discussion_r559353652

Reply:
fixed